ragna/core/_rag.py

```diff
@@ -188,7 +196,17 @@ async def prepare(self) -> Message:
                 detail=RagnaException.EVENT,
             )
 
-        await self._run(self.source_storage.store, self.documents)
+        from ragna.core import Document
+        if type(self.source_storage).__ragna_input_type__ == Document:
```

Comment:
I would prefer something like

```suggestion
        if issubclass(self.source_storage.__ragna_input_type__, Document):
```